### ISSUE TITLE
Add MakeRelative alias to DirectoryPath and FilePath

### ISSUE THREAD

**Issue by augustoproiete** [cake-build/cake#2838]
### What You Are Seeing?

`MakeAbsolute` is a useful alias to transform a relative path into an absolute path based on the current `WorkingDirectory`, but there's not an equivalent alias for doing the opposite, that is, transform an absolute path into a relative path.

### What is Expected?

I'd like Cake to provide a `MakeRelative` alias that transforms an absolute path into a relative path, which is useful when printing messages to the console log, for ease of reading.

e.g.
```csharp
// Environment.WorkingDirectory => "C:/Working/"

var relativePath = MakeRelative("C:/Working/build/MyApp.csproj");
// relativePath should be "./build/MyApp.csproj"
```

### What version of Cake are you using?

0.38.4

---

ps: I'm happy to send a PR that adds this feature.

**Comment by devlead** [comment]
Might be a good edition
Example of code making path relative
https://github.com/cake-build/cake/blob/80d787c5c9c0e102e088ef1a41c0f2e1ae95e790/src/Cake.Common/IO/Zipper.cs#L225-L233

A MakeRelative alias should take rootPath as the last parameter and be optional, that way it could default to working directory but you could also specify a path.

**Comment by patriksvensson** [comment]
@augustoproiete @devlead Better to use the `RelativePathResolver` which was made for this: https://github.com/cake-build/cake/blob/develop/src/Cake.Core/IO/RelativePathResolver.cs

The method above should probably be rewritten to use that as well...

**Comment by devlead** [comment]
Fixed by #2923
